js/plugins/vertexai/src/gemini.ts

```diff
@@ -398,7 +398,10 @@ function fromGeminiFunctionResponsePart(part: GeminiPart): Part {
 }
 
 // Converts vertex part to genkit part
-function fromGeminiPart(part: GeminiPart): Part {
+function fromGeminiPart(part: GeminiPart, jsonMode: boolean): Part {
+  if (jsonMode && part.text !== undefined) {
+    return { data: JSON.parse(part.text) };
```

Comment:
is `JSON.parse` safe here? Do we need to guard agains markdown and extraneous text? Or does the gemini json mode guarantee to ouput clean json only?

Reply:
it's not safe, we need to use lenient parsing, there's an open issue for it. Working on a fix (this is also in main)

Reply:
I think it will guarantee clean json but not complete JSON, but will double check